File: mobile/ios/NoteAI/NoteAIPendingCaptureModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(NoteAIPendingCaptureModule, NSObject)

RCT_EXTERN_METHOD(getPendingCaptureRequest:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(clearPendingCaptureRequest:(NSString *)id
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: mobile/ios/NoteAI/NoteAIPendingCaptureModule.swift
import Foundation
import React

@objc(NoteAIPendingCaptureModule)
class NoteAIPendingCaptureModule: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool {
    false
  }

  @objc(getPendingCaptureRequest:rejecter:)
  func getPendingCaptureRequest(
    _ resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) {
    resolve(NoteAIPendingCaptureStore.loadDictionary() ?? NSNull())
  }

  @objc(clearPendingCaptureRequest:resolver:rejecter:)
  func clearPendingCaptureRequest(
    _ id: String,
    resolver resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) {
    resolve(NoteAIPendingCaptureStore.clear(id: id))
  }
}
